presto-main/src/test/java/io/prestosql/server/security/oauth2/TestingHydraService.java

```diff
@@ -0,0 +1,117 @@
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.startupcheck.OneShotStartupCheckStrategy;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.time.Duration;
+
+import static io.prestosql.util.RandomUtils.randomAlphanumeric;
+
+class TestingHydraService
+        implements Closeable, AutoCloseable
+{
+    static final int TTL_ACCESS_TOKEN_IN_SECONDS = 2;
+    private static final String HYDRA_IMAGE = "oryd/hydra:v1.4.2";
+    private static final String DSN = "postgres://hydra:mysecretpassword@database:5432/hydra?sslmode=disable";
+
+    private final Network network = Network.newNetwork();
+
+    private final PostgreSQLContainer<?> databaseContainer = new PostgreSQLContainer<>()
+            .withNetwork(network)
+            .withNetworkAliases("database")
+            .withUsername("hydra")
+            .withPassword("mysecretpassword")
+            .withDatabaseName("hydra");
+
+    private final GenericContainer<?> migrationContainer = createHydraContainer()
+            .withCommand("migrate sql --yes " + DSN)
+            .withStartupCheckStrategy(new OneShotStartupCheckStrategy().withTimeout(Duration.ofMinutes(5)));
+
+    private final FixedHostPortGenericContainer<?> consentContainer = new FixedHostPortGenericContainer<>("oryd/hydra-login-consent-node:v1.4.2")
+            .withNetwork(network)
+            .withNetworkAliases("consent")
+            .withExposedPorts(3000)
+            .withEnv("HYDRA_ADMIN_URL", "http://hydra:4445")
+            .withEnv("NODE_TLS_REJECT_UNAUTHORIZED", "0")
+            .waitingFor(Wait.forHttp("/").forStatusCode(200));
+
+    private final FixedHostPortGenericContainer<?> hydraContainer = createHydraContainer()
+            .withNetworkAliases("hydra")
+            .withExposedPorts(4444, 4445)
+            .withEnv("SECRETS_SYSTEM", randomAlphanumeric(32))
+            .withEnv("DSN", DSN)
+            .withEnv("URLS_SELF_ISSUER", "http://hydra:4444/")
+            .withEnv("URLS_CONSENT", "http://consent:3000/consent")
+            .withEnv("URLS_LOGIN", "http://consent:3000/login")
+            .withEnv("OAUTH2_ACCESS_TOKEN_STRATEGY", "jwt")
+            .withEnv("TTL_ACCESS_TOKEN", TTL_ACCESS_TOKEN_IN_SECONDS + "s")
+            .withCommand("serve all --dangerous-force-http")
+            .waitingFor(Wait.forHttp("/health/ready").forPort(4444).forStatusCode(200));
+
+    private final AutoCloseableCloser closer = AutoCloseableCloser.create();
+
+    TestingHydraService()
+    {
+        closer.register(network);
+        closer.register(databaseContainer);
+        closer.register(migrationContainer);
+        closer.register(consentContainer);
+        closer.register(hydraContainer);
+    }
+
+    void start()
```

Comment:
These methods look to be part of the public interface of this class, so they should be public